### PR TITLE
Revamp ppdet reader

### PR DESCRIPTION
**Pull request by willthefrog**
this is a total rewrite, main goal is to make the interface user friendly and more flexible , with extra bonus of speeding up some models which are heavy on preprocessing.

this is the first part of the change, config changes are split out to make code review easier

### CODE REVIEW

File: PaddleCV/PaddleDetection/ppdet/utils/voc_eval.py [hunk 1]
```diff
@@ -70,64 +67,31 @@ def bbox_eval(results,
...
+
+        bbox_idx = 0
+        for i, bbox_num in enumerate(bbox_lengths):
+            bbox = bboxes[bbox_idx: bbox_idx + bbox_num]
+            gt_box = gt_boxes[i]
+            gt_label = gt_labels[i]
+            difficult = None if difficults is None else \
+                difficults[i]
+            detection_map.update(bbox, gt_box, gt_label, difficult)
+            bbox_idx += bbox_num
```
Comment (heavengate) [inline PaddleCV/PaddleDetection/ppdet/utils/voc_eval.py]: 这是按

Comment (heavengate) [inline PaddleCV/PaddleDetection/ppdet/utils/voc_eval.py:82]: 1. ssd，rcnn是按LoDTensor读的，这样改跑不通吧
2. 这样是按照zero padding处理？ 这样的话会padding的全0也统计成gtbox，recall会计算的偏低

Comment (willthefrog) [reply PaddleCV/PaddleDetection/ppdet/utils/voc_eval.py:82]: the extra dict contains regular unpadded numpy arrays (not lod tensors), these are already verified to be working correctly.

Comment (willthefrog) [reply PaddleCV/PaddleDetection/ppdet/utils/voc_eval.py:82]: with the new reader, there is no need to pass numpy arrays as variable (a roundtrip to GPU and back), ndarrays not actually used by the graph can be passed in as extra var

**Pull request closed without merging**